test/extended/openstack/dualstack.go

```diff
@@ -0,0 +1,144 @@
+				testDeployment := createTestDeployment("lb-dualstack-dep", labels, 2, protocolUnderTest, 8081)
+				deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(ctx,
+					testDeployment, metav1.CreateOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Creating Openshift LoadBalancer Service")
+				svcName := "lb-default-svc"
+				svcPort := int32(8082)
+				jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+				jig.Labels = labels
+				svc, err := jig.CreateLoadBalancerService(ctx, loadBalancerServiceTimeout, func(svc *v1.Service) {
+					svc.Spec.Ports = []v1.ServicePort{{Protocol: protocolUnderTest, Port: svcPort, TargetPort: intstr.FromInt(8081)}}
+					svc.Spec.Selector = labels
+					svc.Spec.IPFamilyPolicy = (*v1.IPFamilyPolicy)(&ipFamilyPolicy)
+					svc.Spec.IPFamilies = []v1.IPFamily{v1.IPFamily(ipFamiliesList[0]), v1.IPFamily(ipFamiliesList[1])}
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Checks from openshift perspective")
+				loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+				o.Expect(loadBalancerId).ShouldNot(o.BeEmpty(), "load-balancer-id annotation missing")
+				o.Expect(svc.Status.LoadBalancer.Ingress).ShouldNot(o.BeEmpty(), "svc.Status.LoadBalancer.Ingress should not be empty")
+				svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+				o.Expect(svcIp).ShouldNot(o.BeEmpty(), "Ingress IP missing on svc Status")
+				o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeCluster),
+					"Unexpected ExternalTrafficPolicy on svc specs")
+
+				g.By("Checks from openstack perspective")
+				lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(lb.Provider).Should(o.Equal(strings.ToLower(lbProviderUnderTest)), "Unexpected provider in the Openstack LoadBalancer")
+				if isIpv4(lb.VipAddress) { // No FIP assignment on ipv6
+					fip, err := getFipbyFixedIP(networkClient, lb.VipAddress)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(fip.FloatingIP).Should(o.Equal(svcIp), "Unexpected floatingIp in the Openstack LoadBalancer")
+				}
+
+				g.By("accessing the service 100 times from outside and storing the name of the pods answering")
+				results := make(map[string]int)
+				for i := 0; i < 100; i++ {
+					// https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
+					podName, err := getPodNameThroughLb(svcIp, fmt.Sprintf("%d", svcPort), protocolUnderTest, "hostname")
+					if err != nil {
+						e2e.Logf("Error detected while accessing the LoadBalancer service on try %d: %q", i, err)
+					} else {
+						results[podName]++
+					}
+				}
+				e2e.Logf("Pods accessed after 100 requests:\n%v\n", results)
```

Comment:
It depends. What do you want to test exactly?

Reply:
if you don't want to test the lbMethod (I don't think it's needed because it is already tested in loadbalancers.go for both ipv4 and ipv6), you can reduce the number of attempts to 5 for example and check that any pod is accessed at least 1.

Reply:
it's a bit surprising to see this information logged, instead of being used for a test.